airbyte-integrations/connectors/source-salesforce/source_salesforce/rate_limiting.py

```diff
@@ -0,0 +1,72 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+import sys
+
+import backoff
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.sources.streams.http.exceptions import DefaultBackoffException
+from requests import codes, exceptions
+
+TRANSIENT_EXCEPTIONS = (
+    DefaultBackoffException,
+    exceptions.ConnectTimeout,
+    exceptions.ReadTimeout,
+    exceptions.ConnectionError,
+    exceptions.HTTPError,
+)
+
+logger = AirbyteLogger()
+
+
+def default_backoff_handler(max_tries: int, factor: int, **kwargs):
```

Comment:
why create a separate backoff handler instead of using the `backoff_handler` in the CDK?

Reply:
For this connector, it is necessary to slightly change the Rate Limiting logic by adding `exceptions.HTTPError` to `TRANSIENT_EXCEPTIONS ` and updating the `should_give_up `method.